back/engines/commercial/analysis/app/models/analysis/analysis.rb

```diff
@@ -30,12 +30,23 @@ class Analysis < ApplicationRecord
 
     validate :project_xor_phase_present
 
+    def inputs
+      scope = Idea.published
+      if phase_id
+        scope.in_phase(phase_id)
+      elsif project.timeline? # must be ideation
+        scope.where(project_id: project_id, creation_phase: nil)
```

Comment:
The gist is that this thing (either the associated phase or project) is the 'form context', so it needs to be something that has a form attached. In our current (somewhat flawed) model, the ideation form is always attached to the project, the survey form always to the phase. So in case the attached form context is a project, it needs to be ideation (or ideation-like).

I added a few comments to explain. I also started on test coverage and taking a different approach, but it's honestly a lot of work with slow tests as a result, and little value given the way this is going to be used. So I'm going to stick with it for now. Did add some more validations on the validity of the passed project/phase.